nano/core_test/websocket.cpp

```diff
@@ -160,6 +160,72 @@ TEST (websocket, subscription_edge)
 	node1->stop ();
 }
 
+// Test client subscribing to changes in active_difficulty
+TEST (websocket, subscribe_active_difficulty)
+{
+	nano::system system (24000, 1);
+	nano::node_init init1;
+	nano::node_config config;
+	nano::node_flags node_flags;
+	config.websocket_config.enabled = true;
+	config.websocket_config.port = 24078;
+
+	auto node1 (std::make_shared<nano::node> (init1, system.io_ctx, nano::unique_path (), system.alarm, config, system.work, node_flags));
+	node1->start ();
+	system.nodes.push_back (node1);
+
+	ASSERT_EQ (0, node1->websocket_server->subscriber_count (nano::websocket::topic::active_difficulty));
+
+	// Subscribe to active_difficulty and wait for response asynchronously
+	ack_ready = false;
+	auto client_task = ([&node1]() -> boost::optional<std::string> {
+		auto response = websocket_test_call ("::1", "24078", R"json({"action": "subscribe", "topic": "active_difficulty", "ack": true})json", true, true);
+		return response;
+	});
+	auto client_future = std::async (client_task);
```

Comment:
Seems like the problem with travis might be here, can you try forcing the policy? Would be something like:
```cpp
auto client_future = std::async (std::launch::async, client_task);
```

Without forcing, it could fallback to `defer` which makes the test fail. However, forcing the policy can [throw](https://en.cppreference.com/w/cpp/thread/async). If that doesn't work, could do similarly to other tests where a flag is passed by reference to a `std::thread` directly.

Reply:
thanks! fixed now, tests passing